Error while checking build/O2/fullCI for a561783e69c0e16fe8be2d958b5984df2ee1ce59 at 2022-12-02 11:39:
```
## sw/BUILD/O2-latest/log
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:180:156: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:182:101: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:183:169: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:183:171: error: template argument 1 is invalid
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:185:210: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:186:120: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:188:120: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:198:144: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:200:135: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:200:137: error: expected primary-expression before ')' token
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:203:190: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:203:90: error: expected primary-expression before 'const'
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:203:90: error: expected ')' before 'const'
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:207:160: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:207:60: error: expected primary-expression before 'const'
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:207:60: error: expected ')' before 'const'
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:225:140: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:225:30: error: redefinition of 'ROOT::TGenericClassInfo* ROOT::GenerateInitInstanceLocal(const int*)'
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:227:85: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:228:153: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:228:155: error: template argument 1 is invalid
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:230:178: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:231:104: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:233:104: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:243:128: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:243:23: error: redefinition of 'ROOT::TGenericClassInfo* ROOT::GenerateInitInstance(const int*)'
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:245:119: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:245:121: error: expected primary-expression before ')' token
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:248:174: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:248:90: error: expected primary-expression before 'const'
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:248:90: error: expected ')' before 'const'
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:252:144: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:252:60: error: expected primary-expression before 'const'
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:252:60: error: expected ')' before 'const'
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:566:106: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:569:100: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:569:114: error: 'Class_Name' is not a template function
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:575:100: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:575:116: error: 'ImplFileName' is not a template function
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:581:92: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:581:108: error: 'ImplFileLine' is not a template function
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:587:96: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:587:110: error: 'Dictionary' is not a template function
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:594:96: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:594:105: error: 'Class' is not a template function
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:605:90: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:608:84: error: wrong number of template arguments (2, should be 1)
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:608:98: error: 'Class_Name' is not a template function
/sw/BUILD/423540f3067d77b0b5426e40797f12eb89379290/O2/Detectors/TRD/calibration/G__O2TRDCalibration.cxx:614:84: error: wrong number of template arguments (2, should be 1)
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10384/a561783e69c0e16fe8be2d958b5984df2ee1ce59/build_O2_fullCI/pretty.html).

--- a/Detectors/GlobalTracking/CMakeLists.txt
+++ b/Detectors/GlobalTracking/CMakeLists.txt
@@ -21,6 +21,7 @@ o2_add_library(GlobalTracking
                        src/MatchITSTPCQC.cxx
                        src/ITSTPCMatchingQCParams.cxx
                        src/MatchGlobalFwdParam.cxx
+                       src/MatchTOFParams.cxx
   PUBLIC_LINK_LIBRARIES O2::Framework
                         O2::DataFormatsTPC
                         O2::DataFormatsITSMFT

--- a/Detectors/GlobalTracking/include/GlobalTracking/MatchTOF.h
+++ b/Detectors/GlobalTracking/include/GlobalTracking/MatchTOF.h
@@ -40,6 +40,7 @@
 #include "TPCFastTransform.h"
 #include "CommonDataFormat/InteractionRecord.h"
 #include "CorrectionMapsHelper.h"
+#include "GlobalTracking/MatchTOFParams.h"
 
 // from FIT
 #include "DataFormatsFT0/RecPoints.h"
@@ -220,6 +221,9 @@ class MatchTOF
   const o2::globaltracking::RecoContainer* mRecoCont = nullptr;
   o2::InteractionRecord mStartIR{0, 0}; ///< IR corresponding to the start of the TF
 
+  // TOF matching params (work in progress)
+  const MatchTOFParams* mMatchParams = nullptr;
+
   // for derived class
   int mCurrTracksTreeEntry = 0; ///< current tracks tree entry loaded to memory
 
@@ -321,7 +325,7 @@ class MatchTOF
   TStopwatch mTimerMatchITSTPC;
   TStopwatch mTimerMatchTPC;
   TStopwatch mTimerDBG;
-  ClassDefNV(MatchTOF, 4);
+  ClassDefNV(MatchTOF, 5);
 };
 } // namespace globaltracking
 } // namespace o2

--- a/Detectors/GlobalTracking/include/GlobalTracking/MatchTOFParams.h
+++ b/Detectors/GlobalTracking/include/GlobalTracking/MatchTOFParams.h
@@ -1,0 +1,34 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \author fnoferin@cern.ch
+
+#ifndef ALICEO2_MATCHTOF_PARAMS_H
+#define ALICEO2_MATCHTOF_PARAMS_H
+
+#include "CommonUtils/ConfigurableParam.h"
+#include "CommonUtils/ConfigurableParamHelper.h"
+
+namespace o2
+{
+namespace globaltracking
+{
+
+struct MatchTOFParams : public o2::conf::ConfigurableParamHelper<MatchTOFParams> {
+  float calibMaxChi2 = 3.0;
+
+  O2ParamDef(MatchTOFParams, "MatchTOF");
+};
+
+} // namespace globaltracking
+} // end namespace o2
+
+#endif

--- a/Detectors/GlobalTracking/src/GlobalTrackingLinkDef.h
+++ b/Detectors/GlobalTracking/src/GlobalTrackingLinkDef.h
@@ -25,6 +25,9 @@
 #pragma link C++ class o2::globaltracking::MatchCosmicsParams + ;
 #pragma link C++ class o2::conf::ConfigurableParamHelper < o2::globaltracking::MatchCosmicsParams> + ;
 
+#pragma link C++ class o2::globaltracking::MatchTOFParams + ;
+#pragma link C++ class o2::conf::ConfigurableParamHelper < o2::globaltracking::MatchTOFParams> + ;
+
 #pragma link C++ class o2::conf::ConfigurableParamHelper < o2::globaltracking::ITSTPCMatchingQCParams> + ;
 
 #pragma link C++ class o2::globaltracking::GloFwdAssessment + ;

--- a/Detectors/GlobalTracking/src/MatchTOF.cxx
+++ b/Detectors/GlobalTracking/src/MatchTOF.cxx
@@ -57,6 +57,10 @@ ClassImp(MatchTOF);
 //______________________________________________
 void MatchTOF::run(const o2::globaltracking::RecoContainer& inp)
 {
+  if (!mMatchParams) {
+    mMatchParams = &o2::globaltracking::MatchTOFParams::Instance();
+  }
+
   ///< running the matching
   mRecoCont = &inp;
   mStartIR = inp.startIR;
@@ -1160,7 +1164,7 @@ void MatchTOF::selectBestMatches()
       flags = flags | o2::dataformats::CalibInfoTOF::kMultiHit;
     }
 
-    if (matchingPair.getChi2() < 3) { // extra cut in Chi2 for storing calib info
+    if (matchingPair.getChi2() < mMatchParams->calibMaxChi2) { // extra cut in ChiSquare for storing calib info
       mCalibInfoTOF.emplace_back(mTOFClusWork[matchingPair.getTOFClIndex()].getMainContributingChannel(),
                                  mTimestamp / 1000 + int(mTOFClusWork[matchingPair.getTOFClIndex()].getTimeRaw() * 1E-12), // add time stamp
                                  deltat,

--- a/Detectors/GlobalTracking/src/MatchTOF.cxx
+++ b/Detectors/GlobalTracking/src/MatchTOF.cxx
@@ -1160,7 +1160,7 @@ void MatchTOF::selectBestMatches()
       flags = flags | o2::dataformats::CalibInfoTOF::kMultiHit;
     }
 
-    if (matchingPair.getChi2() < 3) { // extra cut in Chi2
+    if (matchingPair.getChi2() < 3) { // extra cut in Chi2 for storing calib info
       mCalibInfoTOF.emplace_back(mTOFClusWork[matchingPair.getTOFClIndex()].getMainContributingChannel(),
                                  mTimestamp / 1000 + int(mTOFClusWork[matchingPair.getTOFClIndex()].getTimeRaw() * 1E-12), // add time stamp
                                  deltat,

--- a/Detectors/GlobalTracking/src/MatchTOFParams.cxx
+++ b/Detectors/GlobalTracking/src/MatchTOFParams.cxx
@@ -1,0 +1,17 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file MatchTOFParams.h
+/// \brief Configurable params for TOF matching
+/// \author fnoferin@cern.ch
+
+#include "GlobalTracking/MatchTOFParams.h"
+O2ParamImpl(o2::globaltracking::MatchTOFParams);

--- a/Detectors/TOF/workflowIO/include/TOFWorkflowIO/DigitReaderSpec.h
+++ b/Detectors/TOF/workflowIO/include/TOFWorkflowIO/DigitReaderSpec.h
@@ -43,6 +43,7 @@ class DigitReader : public Task
   int mState = 0;
   int mCurrentEntry = 0;
   bool mUseMC = true;
+  int mDelayInMuSec1TF = 0;
   WindowFiller mFiller;
   std::unique_ptr<TFile> mFile = nullptr;
   std::vector<o2::tof::Digit> mDigits, *mPdigits = &mDigits;

--- a/Detectors/TOF/workflowIO/src/DigitReaderSpec.cxx
+++ b/Detectors/TOF/workflowIO/src/DigitReaderSpec.cxx
@@ -35,6 +35,8 @@ void DigitReader::init(InitContext& ic)
   LOG(debug) << "Init Digit reader!";
   auto filename = o2::utils::Str::concat_string(o2::utils::Str::rectifyDirectory(ic.options().get<std::string>("input-dir")),
                                                 ic.options().get<std::string>("tof-digit-infile"));
+  mDelayInMuSec1TF = atoi(ic.options().get<std::string>("delay-1st-tf").c_str()) * 1E6;
+
   mFile.reset(TFile::Open(filename.c_str()));
   if (!mFile->IsOpen()) {
     LOG(error) << "Cannot open the " << filename.c_str() << " file !";
@@ -46,6 +48,12 @@ void DigitReader::init(InitContext& ic)
 
 void DigitReader::run(ProcessingContext& pc)
 {
+  static bool firstCall = true;
+  if (firstCall) {
+    usleep(mDelayInMuSec1TF);
+  }
+  firstCall = false;
+
   if (mState != 1) {
     return;
   }
@@ -118,6 +126,7 @@ DataProcessorSpec getDigitReaderSpec(bool useMC)
     AlgorithmSpec{adaptFromTask<DigitReader>(useMC)},
     Options{
       {"tof-digit-infile", VariantType::String, "tofdigits.root", {"Name of the input file"}},
+      {"delay-1st-tf", VariantType::String, "none", {"delay in seconds before 1st TF"}},
       {"input-dir", VariantType::String, "none", {"Input directory"}}}};
 }
 

--- a/Detectors/TOF/workflowIO/src/DigitReaderSpec.cxx
+++ b/Detectors/TOF/workflowIO/src/DigitReaderSpec.cxx
@@ -35,7 +35,7 @@ void DigitReader::init(InitContext& ic)
   LOG(debug) << "Init Digit reader!";
   auto filename = o2::utils::Str::concat_string(o2::utils::Str::rectifyDirectory(ic.options().get<std::string>("input-dir")),
                                                 ic.options().get<std::string>("tof-digit-infile"));
-  mDelayInMuSec1TF = atoi(ic.options().get<std::string>("delay-1st-tf").c_str()) * 1E6;
+  mDelayInMuSec1TF = atof(ic.options().get<std::string>("delay-1st-tf").c_str()) * 1E6;
 
   mFile.reset(TFile::Open(filename.c_str()));
   if (!mFile->IsOpen()) {

